pyquil/api/_quantum_computer.py

```diff
@@ -669,45 +671,114 @@ def get_qc(name: str, *, as_qvm: bool = None, noisy: bool = None,
 def local_qvm() -> Iterator[Tuple[subprocess.Popen, subprocess.Popen]]:
     """A context manager for the Rigetti local QVM and QUIL compiler.
 
+    .. deprecated:: 2.11
+        Use py:func:`local_forest_runtime` instead.
+    """
+    warnings.warn(DeprecationWarning("Use of pyquil.api.local_qvm has been deprecated.\n"
+                                     "Please use pyquil.api.local_forest_runtime instead."))
+    with local_forest_runtime() as (qvm, quilc):
+        yield (qvm, quilc)
+
+
+def _port_used(host: str, port: int):
+    """Check if a (TCP) port is listening.
+
+    :param host: Host address to check.
+    :param port: TCP port to check.
+
+    :returns: ``True`` if a process is listening on the specified host/port, ``False`` otherwise
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.connect((host, port))
+        return True
+    except ConnectionRefusedError:
+        return False
+    finally:
+        s.close()
+
+
+@contextmanager
+def local_forest_runtime(
+        *,
+        host: str = '127.0.0.1',
+        qvm_port: int = 5000,
+        quilc_port: int = 5555,
+        use_protoquil: bool = False) -> Iterator[Tuple[subprocess.Popen, subprocess.Popen]]:
+    """A context manager for local QVM and QUIL compiler.
+
     You must first have installed the `qvm` and `quilc` executables from
     the forest SDK. [https://www.rigetti.com/forest]
 
-    This context manager will start up external processes for both the
-    compiler and virtual machine, and then terminate them when the context
-    is exited.
+    This context manager will ensure that the designated ports are not used, start up `qvm` and
```

Comment:
@notmgsk Documented the new behaviour and the tuple values here. I wasn't sure if I should use `:return:` since this is an iterator.

Looking forward to your feedback!

Reply:
If it's not standard to use `:return:` for an iterator, then what we have now is fine.

Reply:
Added a return section, resolving this